src/util/address.rs

```diff
@@ -727,6 +727,23 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Determine if an address is related to a given secp256k1 public key
+    pub fn is_related_to_secp256k1_key(&self, pubkey: &PublicKey) -> bool {
+        let pubkey_hash = pubkey.pubkey_hash();
+        let redeem_hash = segwit_redeem_hash(&pubkey_hash);
+        let payload = self.as_payload_bytes();
+        (*pubkey_hash == *payload) || (*redeem_hash == *payload)
+    }
+
+    /// Pull out payload as byte array
+    pub fn as_payload_bytes(&self) -> &[u8] {
```

Comment:
I'm not sure if we should make this public. Anyway I think the name `payload_as_bytes` is more natural. (non-native speaker, though)

Reply:
I agree with this change

Reply:
I changed the name, but left this as public.

I have no strong feelings. I will change this however is desired